baml_language/tools/aws-systest/run-e2e.sh

```
@@ -0,0 +1,360 @@
+  echo "Use --profile boundaryml-dev (or any profile that maps to $DEV_ACCOUNT)."
+  exit 1
+fi
+echo "AWS account: $ACCOUNT (profile $PROFILE, region $REGION) — boundaryml-dev guard OK"
+
+if [ "$SKIP_BUILD" -eq 0 ]; then
+  log "Build harness (native$([ "$RUN_CLOUD" -eq 1 ] && echo ' + musl'))"
+  (cd "$SCRIPT_DIR" && cargo build --release) || { record "build:native" 1 "cargo build failed"; }
+  [ -x "$NATIVE_BIN" ] && record "build:native" 0 "ok"
+  if [ "$RUN_CLOUD" -eq 1 ]; then
+    (cd "$SCRIPT_DIR" && cargo zigbuild --release --target "$MUSL_TARGET") \
+      && record "build:musl" 0 "static ELF" || record "build:musl" 1 "zigbuild failed"
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)